[NEW] Allow to send Agent custom fields through the Omnichannel CRM integration

renatobecker opened the pull request:
Some integrations need more information about the `Agent`, in addition to the default fields sent through the Omnichannel CRM integration.
To allow this, a new field(`sendToIntegrations`) will be supported in the `custom fields` of the regular Users.

The CRM integration will send information from the `custom fields` that have the field 
```
{
...
    sendToIntegrations: true,
...
}
```

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RocketChat/Rocket.Chat?pullRequest=16286) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/RocketChat/Rocket.Chat?pullRequest=16286) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RocketChat/Rocket.Chat?pullRequest=16286) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/RocketChat/Rocket.Chat?pullRequest=16286) <br/>All committers have signed the CLA.

sampaiodiego commented on app/livechat/server/lib/Livechat.js:
```diff
@@ -603,6 +603,17 @@ export const Livechat = {
+			try {
+				const customFields = JSON.parse(settings.get('Accounts_CustomFields'));
+				return Object.keys(customFields)
+					.filter((customFieldKey) => customFields[customFieldKey].allowExternalSharing === true)
```
we need to document the property `allowExternalSharing` right? so people can know it exists and use it too..

also, what about changing the name to make it more clear what it will cause, I thought something like `sendToIntegrations`?

renatobecker commented:
@sampaiodiego I changed the name of the property according your suggestion, it makes sense to me.
I have, also, added the related documentation as you can see [here](https://github.com/RocketChat/docs/pull/1693).

Thanks.

sampaiodiego commented on app/livechat/server/lib/Livechat.js:
```diff
@@ -627,6 +627,17 @@ export const Livechat = {
+			try {
+				const customFields = JSON.parse(settings.get('Accounts_CustomFields'));
+				return Object.keys(customFields)
+					.filter((customFieldKey) => customFields[customFieldKey].sendToIntegrations === true)
+					.map((field) => field);
```
I'm sorry, I didn't notice this before, but that `.map` isn't doing anything right? 

```suggestion
					.filter((customFieldKey) => customFields[customFieldKey].sendToIntegrations === true);
```

renatobecker replied:
@sampaiodiego you're right, there was no reason for that `.map` method, my bad.
I just removed it, thanks.
